### PR TITLE
[TACO-2309] Fix Activity model and make initializer public

### PR DESCRIPTION
**Pull request by jakeheis**
This will allow clients to create `Activity` objects which then makes it posslbe to search for specific activities using `UserWorkoutFileSearch.activities`. The model did not match what is expected per docs so the model itself was also fixed.

The `Activity` struct is also used in `SearchInputReport` so there's a chance for regression there, but it looks to me in the docs that this updated `Activity` shape is what is actually expected in those API calls

### CODE REVIEW

File: Zone5/Activities/Activity.swift [hunk 1]
```diff
@@ -6,23 +6,24 @@ public struct Activity: Searchable, Hashable {
 	public var id: Int
 
 	/// The type of activity this activity's `id` is related to.
-	public var activity: ActivityResultType?
-
-	/// The sport related to this activity.
-	public var type: ActivityType?
+	public var type: ActivityResultType?
+
```
Comment (jeanhall) [inline Zone5/Activities/Activity.swift]: Funny, I was just looking at this yesterday and noted that it was wrong. 

Yes, you are correct, the ActivityResultType field should be called type. There is also an ActivityType field which should be called sport. So both fields had incorrect names.

Can you please update the PR to add in
`public var sport: ActivityType?`

Thanks!

**Review by jeanhall — CHANGES_REQUESTED** [review]

File: Zone5/Activities/Activity.swift (hunk 1 above)
Comment (jakeheis) [reply Zone5/Activities/Activity.swift]: Updated :+1:

**Review by charliebartel — APPROVED** [review]

**Review by jeanhall — APPROVED** [review]
Thanks! Looks good to me.

**Comment by jakeheis** [comment]
@charliebartel version bumped in 1c26f45e32261537c03f530a394c56fa37f92e30

**Review by tww0003 — APPROVED** [review]

**Pull request merged**